### PR TITLE
Wip 9622 math p invalid data ns

### PR DESCRIPTION
**Pull request by nshehz**

### CODE REVIEW

File: device/src/Device.cc [hunk 1]
```diff
@@ -81,8 +81,10 @@ namespace ChimeraTK {
 
   bool Device::isFunctional() const {
     if(static_cast<bool>(_deviceBackendPointer) != false) {
+      std::cout<<"Device::isFunctional():"<< _deviceBackendPointer->isFunctional()<<std::endl;
```
Comment (killenb) [inline device/src/Device.cc:84]: Please remove leftover debug output.

File: device_backends/LogicalNameMapping/src/LNMMathPlugin.cc [hunk 2]
```diff
@@ -336,14 +336,22 @@ namespace ChimeraTK { namespace LNMBackend {
...
+      }
+    }
 
     // evaluate the expression and store into application buffer
     h.computeResult(_target->accessChannel(0), buffer_2D[0]);
 
     // update version number and validity from target
     this->_versionNumber = _target->getVersionNumber();
     this->_dataValidity = _target->dataValidity();
+    if (paramDataValidity == ChimeraTK::DataValidity::faulty) this->_dataValidity = ChimeraTK::DataValidity::faulty;
```
Comment (killenb) [inline device_backends/LogicalNameMapping/src/LNMMathPlugin.cc:354]: Coding style: always put curly braces after if statements.

File: device_backends/NumericAddressedBackend/src/MapFileParser.cpp [hunk 3]
```diff
@@ -334,12 +334,17 @@ namespace ChimeraTK {
 
   void MapFileParser::handle2DNewStyle(const ParsedLine& pl) {
     // search for sequence entries matching the given register, create ChannelInfos from them
+    
+    std::cout<<"value.handle2DNewStyle:"<<std::endl;
```
Comment (killenb) [inline device_backends/NumericAddressedBackend/src/MapFileParser.cpp:338]: Remove leftover debug output

File: device_backends/NumericAddressedBackend/src/MapFileParser.cpp [hunk 4]
```diff
@@ -409,7 +414,9 @@ namespace ChimeraTK {
     while(true) {
       auto it = parsedLinesMap.find(makeSequenceName(pl.pathName, channelLines.size()));
       if(it == parsedLinesMap.end()) break;
+       std::cout<<"it->second.address:"<<it->second.address<<"::pl.address:"<<pl.address<<std::endl;
       if(it->second.address < pl.address)
+
```
Comment (killenb) [inline device_backends/NumericAddressedBackend/src/MapFileParser.cpp:419]: Avoid unnecessary whitespace changes

File: device_backends/include/TransferElement.h [hunk 5]
```diff
@@ -475,15 +475,19 @@ namespace ChimeraTK {
     void postRead(TransferType type, bool updateDataBuffer) {
       // only delegate to doPostRead() the first time postRead() is called in a row.
       if(readTransactionInProgress) {
+        //std::cout<<"TransferElement::postRead:readTransactionInProgress "<<readTransactionInProgress<<std::endl;
```
Comment (killenb) [inline device_backends/include/TransferElement.h:478]: don't leave commented code lines

File: device_backends/include/TransferElement.h [hunk 6]
```diff
@@ -475,15 +475,19 @@ namespace ChimeraTK {
...
+        //std::cout<<"TransferElement::postRead:done "<<std::endl;
+
       }
 
       // Throw on each call of postRead(). All high-level elements for a shared low-level transfer element must see the
       // exception. Note: doPostRead can throw an exception, but in that case _activeException must be false (we can
       // only have one exception at a time). In case other code is added here later which needs to be executed after
       // doPostRead() always, a try-catch block may be necessary.
       if(_activeException) {
+        std::cout<<"TransferElement::postRead rethorwing because exception is active:"<<std::endl;
```
Comment (killenb) [inline device_backends/include/TransferElement.h:490]: remove leftover debug output

**Review by killenb — CHANGES_REQUESTED** [review]
There is some debug output left. I think Device.cc, MapFileParser.cpp and TransferElement.cpp can simply be reverted to the previous state.

**Review by nshehz — COMMENTED** [review]
I have reverted the files which unintentionally got committed and add braces for single line if.

**Review by killenb — APPROVED** [review]
Looks good. Please squash when merging into the master.
